### PR TITLE
KAZOO-5816 handle media.webrtc based on protocol

### PR DESCRIPTION
**Pull request by lazedo**

### CODE REVIEW

File: applications/ecallmgr/src/ecallmgr_fs_xml.erl [hunk 1]
```diff
@@ -468,8 +468,21 @@ check_dtmf_type(Props) ->
...
+build_leg_vars([]) -> [];
+build_leg_vars([_|_]=Prop) -> lists:foldr(fun get_channel_vars/2, [], Prop);
+build_leg_vars(JObj) -> build_leg_vars(kz_json:to_proplist(JObj)).
+
 -spec get_leg_vars(kz_json:object() | kz_term:proplist()) -> iolist().
 get_leg_vars([]) -> [];
+get_leg_vars([Binary|_]=Binaries)
+  when is_binary(Binary) ->
+    ["[^^", ?BRIDGE_CHANNEL_VAR_SEPARATOR
+    ,string:join([kz_term:to_list(V) || V <- Binaries]
```
Comment (jamesaimonetti) [inline applications/ecallmgr/src/ecallmgr_fs_xml.erl]: Did kz_binary:join/2 not work?

Comment (lazedo) [reply applications/ecallmgr/src/ecallmgr_fs_xml.erl]: @jamesaimonetti preserved the way it was already doing in the existing clause. if i change here, then would also have to change the others for consistency. i know, i'm lazy :)

File: applications/ecallmgr/src/ecallmgr_registrar.erl [hunk 2]
```diff
@@ -1295,16 +1318,17 @@ breakup_contact(Contact) when is_binary(Contact) ->
 breakup_contact(Contact) ->
     breakup_contact(kz_term:to_binary(Contact)).
 
--spec proxy_path(kz_term:api_binary(), kz_term:api_binary(), kz_term:api_integer()) -> binary().
-proxy_path('undefined', 'undefined', 'undefined') -> <<>>;
-proxy_path('undefined', 'undefined', Port) -> proxy_path('undefined', <<>>, Port);
-proxy_path('undefined', IP, 'undefined') -> IP;
-proxy_path('undefined', IP, Port) -> <<IP/binary, ":", (kz_term:to_binary(Port))/binary>>;
-proxy_path(Proxy, _, Port) ->
+-spec proxy_path(kz_term:api_binary(), kz_term:api_binary(), kz_term:api_integer(), kz_term:api_binary()) -> binary().
```
Comment (k-anderson) [inline applications/ecallmgr/src/ecallmgr_registrar.erl]: @lazedo won't this result in a websocket request between FS and Kamailio?

Comment (lazedo) [reply applications/ecallmgr/src/ecallmgr_registrar.erl]: @k-anderson `proxy_path` is only used for displaying

**Pull request merged**